### PR TITLE
Rename some options (backwards-compatible) and add new `defaultProtocol` option

### LINKED ISSUE #70: Ability to normalize protocol to HTTPS by default

### PR DESCRIPTION
**Pull request by szmarczak**
Fixes #70

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,17 +1,18 @@
 'use strict';
 // TODO: Use the `URL` global when targeting Node.js 10
-const URLParser = typeof URL === 'undefined' ? require('url').URL : URL;
+const URL = require('url').URL; // eslint-disable-line prefer-destructuring
```
Comment (sindresorhus) [inline index.js]: This needs to stay like it is. See: https://github.com/sindresorhus/normalize-url/commit/70b5afb59a354278d98f5035067fb00dfcc4ba89

File: index.js [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 
 function testParameter(name, filters) {
 	return filters.some(filter => filter instanceof RegExp ? filter.test(name) : filter === name);
 }
 
 module.exports = (urlString, opts) => {
 	opts = Object.assign({
+		defaultProtocol: 'http:',
```
Comment (sindresorhus) [inline index.js]: I think it should be specified without the `:`.

File: index.js [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
 		normalizeProtocol: true,
-		normalizeHttps: false,
-		normalizeHttp: false,
-		stripFragment: true,
+		forceHttp: false,
+		forceHttps: false,
+		stripHash: true,
```
Comment (sindresorhus) [inline index.js]: I'm ok with this renaming, but make it backward compatible by adding some aliases to the old names below. I don't want to do a major release just for this. I just recently did a major release.

File: index.js (hunk 2 above)
Comment (szmarczak) [reply index.js]: I'd leave it with the colon. Argument:

```js
> (new URL('https://sindresorhus.com')).protocol
"https:"
```

File: index.js (hunk 3 above)
Comment (szmarczak) [reply index.js]: :crying_cat_face: Ok... :P

Comment (szmarczak) [reply index.js]: Done.

File: index.js [hunk 4]
```diff
@@ -8,43 +8,53 @@ function testParameter(name, filters) {
...
 	}, opts);
 
+	// Backwards compatibility
+	if (opts.normalizeHttps) {
+		opts.forceHttp = true;
+	}
+
+	if (opts.normalizeHttp) {
+		opts.forceHttps = true;
+	}
```
Comment (sindresorhus) [inline index.js]: `stripHash` too

Comment (szmarczak) [reply index.js]: kk

Comment (szmarczak) [reply index.js]: done

File: index.js [hunk 5]
```diff
@@ -8,43 +8,57 @@ function testParameter(name, filters) {
...
+
+	if (opts.skipFragment) {
+		opts.skipHash = true;
+	}
```
Comment (sindresorhus) [inline index.js]: It's `strip`, not `skip`.

Can you add back at least one test for each of these so we can ensure backwards compatibility is preserved?

Comment (szmarczak) [reply index.js]: done

**Pull request merged**